config/rbac/role.yaml

```diff
@@ -162,6 +178,26 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
+  - cluster.x-k8s.io
+  resources:
+  - machinesets
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - cluster.x-k8s.io
+  resources:
+  - machinesets
```

Comment:
do we need this `machinesets` here?

Reply:
also this seems to be granted here already https://github.com/kubernetes-sigs/cluster-api/pull/5191/files#diff-6d4c4e29ec096149781536bb9118fde8242c1901df764e3eda3019ac8503d6c7L169?

Reply:
It is redundant. I think it's a bug in controller-tools. It seems to add the permissions multiple times when the kubebuilder go tag is not 100% the same. I think the only thing I can do in our code to fix this is to drop the kubebuilder tag oder sync it with others. 

Apart from that I can open an issue in the controller-tools repo.

Reply:
I'll open the issue tomorrow.

Reply:
Issue is open: https://github.com/kubernetes-sigs/controller-tools/issues/612

In theory we could solve it by creating a separate tag for every resource. But I given the tags we currently have this would lead to a lot of separate tags